src/api.ts

```diff
@@ -65,6 +70,51 @@ export const resolvers = {
     switchPerforceChangelist: async (parent: any, args: {changelistId: string}): Promise<OperationStatus> => {
       const perforceManager = getPerforceManagerSingleton();
       return perforceManager ? (await perforceManager.switchChangelist(args.changelistId)) : { isSuccess: false, reason: "Perforce not initialized"};
+    },
+    getGitRepo: async (parent: any, args: {sources: [{repoUrl: string, commit: string}]},
+                       context: { onAddRepoRequest: onAddRepoRequestHandler }): Promise<OperationStatus> => {
+      const subDirs = fs.readdirSync(GIT_ROOT);
+      // Delete the folder if it's too big
+      if ((await isGitFolderBiggerThanMaxSize())) {
```

Comment:
I ran this logic locally on my dev folder.  
It took 130s to scan a total of 14GB, CPU was at ~60% during that time and disk at ~50MB/s read.  
I don't think this is a good approach @ElDuderinos